pkg/cvo/sync_worker.go

```diff
@@ -353,20 +359,23 @@ func (w *SyncWorker) syncPayload(ctx context.Context, work *SyncWork, reporter S
 		})
 		klog.V(2).Infof("Payload loaded from %s with hash %s", desired.Image, payloadUpdate.ManifestHash)
 	}
-	return nil
+	return implicitlyEnabledCaps, nil
 }
 
 // loadUpdatedPayload retrieves the image. If successfully retrieved it updates payload otherwise it returns an error.
-func (w *SyncWorker) loadUpdatedPayload(ctx context.Context, work *SyncWork, cvoOptrName string) error {
+func (w *SyncWorker) loadUpdatedPayload(ctx context.Context, work *SyncWork,
+	cvoOptrName string) ([]configv1.ClusterVersionCapability, error) {
 
 	// reporter hides status updates that occur earlier than the previous failure,
 	// so that we don't fail, then immediately start reporting an earlier status
 	reporter := &statusWrapper{w: w, previousStatus: w.status.DeepCopy()}
-	if err := w.syncPayload(ctx, work, reporter); err != nil {
+
+	implicitlyEnabledCaps, err := w.syncPayload(ctx, work, reporter)
```

Comment:
This is fine, But I am wondering if we should call CheckPayloadCapabilities directly here to get the `implicitlyEnabledCaps`

Reply:
I did it in `syncPayLoad` where I have easy (only?) access to tyhe current payload and the update payload.